src/neuroconv/tools/nwb_helpers/_configuration_models/_hdf5_dataset_io.py

```diff
@@ -78,3 +80,28 @@ def get_data_io_kwargs(self) -> Dict[str, Any]:
             compression_bundle = dict(compression=self.compression_method, compression_opts=compression_opts)
 
         return dict(chunks=self.chunk_shape, **compression_bundle)
+
+    @classmethod
+    def from_existing_neurodata_object(
+        cls, neurodata_object: Container, dataset_name: Literal["data", "timestamps"]
+    ) -> Self:
+        location_in_file = _find_location_in_memory_nwbfile(neurodata_object=neurodata_object, field_name=dataset_name)
+        full_shape = getattr(neurodata_object, dataset_name).shape
+        dtype = getattr(neurodata_object, dataset_name).dtype
+        chunk_shape = getattr(neurodata_object, dataset_name).chunks
+        buffer_shape = getattr(neurodata_object, dataset_name).maxshape
+        compression_method = getattr(neurodata_object, dataset_name).compression
+        compression_opts = getattr(neurodata_object, dataset_name).compression_opts
+        compression_options = dict(compression_opts=compression_opts)
```

Comment:
Huh, but we already have the generic method name and code for applying to in-memory rather than on-disk: https://github.com/catalystneuro/neuroconv/blob/c33dfbfbb367bf978e99a85ce3130383233eeca4/src/neuroconv/tools/nwb_helpers/_configuration_models/_base_dataset_io.py#L248

So what about making this a simpler router like so?

```

        if is_object_dataset_written_to_disk:
            location_in_file = _find_location_in_memory_nwbfile(neurodata_object=neurodata_object, field_name=dataset_name)
            full_shape = getattr(neurodata_object, dataset_name).shape
            dtype = getattr(neurodata_object, dataset_name).dtype
            chunk_shape = getattr(neurodata_object, dataset_name).chunks
            buffer_shape = getattr(neurodata_object, dataset_name).maxshape
            compression_method = getattr(neurodata_object, dataset_name).compression
            compression_opts = getattr(neurodata_object, dataset_name).compression_opts
            compression_options = dict(compression_opts=compression_opts)
        else:
            super().from_neurodata_object(...)
            
        return ...
```